daemon/cmd/hubble.go

```diff
@@ -332,3 +341,24 @@ func (d *Daemon) LookupSecIDByIP(ip net.IP) (id ipcache.Identity, ok bool) {
 func (d *Daemon) GetK8sStore(name string) k8scache.Store {
 	return d.k8sWatcher.GetStore(name)
 }
+
+// getHubbleEventBufferCapacity returns the user configured capacity for
+// Hubble's events buffer. The deprecated flag hubble-flow-buffer-size is
+// evaluated if greater than 0, otherwise the new flag
+// hubble-event-buffer-capacity is used instead.
+func getHubbleEventBufferCapacity(logger logrus.FieldLogger) (container.Capacity, error) {
+	// check deprecated old flag for compatibility
+	// TODO: remove support for HubbleFlowBufferSize once 1.11 is out
+	if option.Config.HubbleFlowBufferSize > 0 {
+		logger.Warningf("Option '%s' is deprecated and will be removed in Cilium 1.11", option.HubbleFlowBufferSize)
+		c, err := container.NewCapacity(option.Config.HubbleFlowBufferSize)
+		if err == nil {
+			return c, nil
+		}
+		// old flag behavior was to silently round up the buffer capacity to a
+		// valid value (eg: 1500 -> 2047, 5000 -> 8191, etc) so adjust provided
+		// value to the nearest valid one for compatibility purpose
+		return container.NewCapacity(int(math.GetMask(math.MSB(uint64(option.Config.HubbleFlowBufferSize)))))
```

Comment:
The value returned by `math.GetMask` is correct, but it reads a bit strange to use a mask as a capacity. I would either use `(1 << math.MSB(...)) - 1` or the following function (from [bit-twiddling hacks](https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2)):

```go
func NextPowerOfTwo(x uint64) uint64 {
	x--
	x |= x >> 1
	x |= x >> 2
	x |= x >> 4
	x |= x >> 8
	x |= x >> 16
	x |= x >> 32
	x++
	return x
}
```

Reply:
A power of 2 minus 1 is effectively a mask of one's so this didn't seem strange to me (well, I wrote that code so that's probably why :smile: ). I'm OK to change this for `(1 << math.MSB(...)) - 1` as you suggest if you think it's more clear but I'm reluctant to introduce a new function for a piece of code that is there only for compatibility reason and set to be removed in the following version of Cilium